optimism/src/mips/interpreter.rs

```diff
@@ -784,8 +785,33 @@ pub fn interpret_rtype<Env: InterpreterEnv>(env: &mut Env, instr: RTypeInstructi
         RTypeInstruction::JumpAndLinkRegister => (),
         RTypeInstruction::SyscallMmap => (),
         RTypeInstruction::SyscallExitGroup => (),
+        RTypeInstruction::SyscallReadHint => (),
         RTypeInstruction::SyscallReadPreimage => (),
-        RTypeInstruction::SyscallReadOther => (),
+        RTypeInstruction::SyscallReadOther => {
+            let fd_id = env.read_register(&Env::constant(4));
```

Comment:
My personal preference is that we don't define a constant that's local and used once in one place. This constant 4 is 4 because the read syscall reads register 4 to find out the FD that it's reading from.
```rust
const REGISTER_FD_ID: u32 = 4;
let fd_id = env.read_register(&Env::constant(REGISTER_FD_ID));
```
is just forcing us to write the same thing 3 times instead of 1.

Reply:
Okay if it only gets used once. But if it becomes something we write over and over again it could start making sense.